doc/Secciones/04_Implementacion.tex

```diff
@@ -0,0 +1,125 @@
+
+\subsection{Inicio del desarrollo}
+Antes de lanzarse a programar hay que hacer una serie de análisis previos y resolver ciertas cuestiones algunas de ellas son:
+\begin{enumerate}
+    \item ¿Para que plataforma está orientado el software?
+    \item ¿En que lenguaje se realizará?
+    \item ¿Se usará una herramienta de control de versiones?
+\end{enumerate}
+Este paso en la planificación es extremadamente importante, ya que determinará el marco de desarrollo bajo un lenguaje específico que no se puede escoger a la primera de cambio ya que cada uno cuenta con una serie de ventajas y desventajas a tener en cuenta en el proyecto. Aunque estas cuestiones serán resueltas en una sección posterior.
+
+\subsection{Encontrar recetas}
+Este es el primer objetivo del software a desarrollar, que permitirá ayudar a los usuarios definidos anteriormente a encontrar recetas que contengan los ingredientes con los que cuentan el usuario. Además, en este punto es necesario definir que tipo de ingredientes son los que tiene cualquier persona en su casa, como por ejemplo: Agua, sal o aceite. Y conforme avance el desarrollo de este objetivo habrá que resolver ciertas cuestiones, como la anterior, que vayan surgiendo. 
+
+\subsection{Despliegue de la solución}
+Para permitir al usuario final el uso de la solución ya sea una aplicación, biblioteca o funcionalidad, debe desplegarse una versión final de la solución, analizando los posibles despliegues dependiendo del tipo de solución que se diseñe a lo largo del proyecto.
+
+\subsection{Sustituir ingredientes}
+También se planifica una de las posibles mejoras a seguir una vez se haya completado el desarrollo del proyecto. En este caso se permitiría que se recomendasen recetas a las que les falta uno o varios ingredientes pero es posible sustituirlos de manera coherente con otros ingredientes con los que sí cuenta el usuario. Pero esto es solo una posible mejora para el futuro del software, teniendo que analizar la viabilidad una vez completado el objetivo anterior. 
+
+\section{Lenguaje del proyecto}
+Para el desarrollo del proyecto, ya sea una plataforma web o una biblioteca, es necesario definir un lenguaje. Aunque no conste como requisito imprescindible, debe adecuarse a la habilidad del equipo de desarrollo, con el objetivo de una robustez y eficiencia en la codificación de la solución al problema planteado. 
+
+Como pequeño análisis previo a justificar el lenguaje de programación a usar, se estiman los siguientes requisitos:
+\begin{enumerate}
+    \item Escalabilidad. Debe ser capaz de manejar el rendimiento requerido por la aplicación, así como el uso por parte de multitud de usuarios de manera concurrente.
```

Comment:
De verdad, ¿por qué ponéis esto? Ya en IV os dije que no tenía sentido. *Todos* los lenguajes son escalables, *todos* son compatibles, *todos* están integrados. Si lo vas a usar porque tienes familiaridad con algún framework o porque tiene herramientas para comprobar la callidad que tengas en mente o por lo que sea, pon exactamente eso. En general, justifica las decisiones por las razones *reales*, no porque pienses que el tutor o el tribunal quiere leer cosas que no tienen sentido.

Reply:
Pero ¿entiendes que es una justificación real? ¿Tú has medido la "escalabilidad" (sea lo que sea) de Java y has visto que era peor que la de Python? ¿No? Pues no digas nada del estilo.

Reply:
Se ha cambiado el apartado de la elección del lenguaje dejando la justificación solo de Python